pkg/natdiscovery/remote_endpoint.go

```diff
@@ -0,0 +1,138 @@
+/*
+© 2021 Red Hat, Inc. and others.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package natdiscovery
+
+import (
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/submariner-io/submariner/pkg/types"
+)
+
+type endpointState int
+
+const (
+	testingPrivateAndPublicIPs endpointState = iota
+	waitingForResponse
+	selectedPublicIP
+	selectedPrivateIP
+	recheckTime                    = 2 * time.Second
+	totalTimeout                   = 60 * time.Second
+	publicToPrivateFailoverTimeout = 1 * time.Second
+)
+
+type remoteEndpointNAT struct {
+	Endpoint               *types.SubmarinerEndpoint
+	state                  endpointState
+	lastCheck              time.Time
+	lastTransition         time.Time
+	started                time.Time
+	UseNAT                 bool
+	UseIP                  string
+	lastPublicIPRequestID  uint64
+	lastPrivateIPRequestID uint64
+}
+
+func newRemoteEndpointNAT(endpoint *types.SubmarinerEndpoint) *remoteEndpointNAT {
+	return &remoteEndpointNAT{
+		Endpoint:       endpoint,
+		state:          testingPrivateAndPublicIPs,
+		started:        time.Now(),
+		lastTransition: time.Now(),
+	}
+}
+
+func (rnat *remoteEndpointNAT) transitionToState(newState endpointState) {
```

Comment:
To keep receiver names to 1 or 2 chars.

```suggestion
func (r *remoteEndpointNAT) transitionToState(newState endpointState) {
```

Reply:
Done.